### PR TITLE
Add upgrade timeout to commander helm release23

### PR DESCRIPTION
**Pull request by aliotta**
Add upgrade timeout to helm chart

### CODE REVIEW

File: tests/test_astronomer_commander.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from tests.helm_template_generator import render_chart
```
Comment (aliotta) [inline tests/test_astronomer_commander.py:1]: These work locally at least. Not sure if we want to get them into the ci conf

File: tests/test_astronomer_commander.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    envVarNames = jmespath.search("spec.template.spec.containers[0].env[*].name", doc)
+    assert "COMMANDER_UPGRADE_TIMEOUT" in envVarNames
+    upgradeTimeoutIndex = envVarNames.index("COMMANDER_UPGRADE_TIMEOUT")
+    assert (
+        doc["spec"]["template"]["spec"]["containers"][0]["env"][upgradeTimeoutIndex][
+            "value"
+        ]
+        == "300"
+    )
```
Comment (danielhoherd) [inline tests/test_astronomer_commander.py:32]: Do you think this is easier to read if we reshape the `env` list using a dict comprehension?
```suggestion
    assert len(doc["spec"]["template"]["spec"]["containers"]) == 1
    env_vars = {x['name']: x['value'] for x in doc["spec"]["template"]["spec"]["containers"][0]['env']}
    assert env_vars['COMMANDER_UPGRADE_TIMEOUT'] == "300"
```

File: tests/test_astronomer_commander.py (hunk 1 above)
Comment (danielhoherd) [reply tests/test_astronomer_commander.py:1]: I've actually [started work on backporting pytest to 0.23](https://github.com/astronomer/astronomer/commit/3cd724f8afcd80ccbe0aab4daca5bb7ad6646d4c). Let's worry about that in the other branch, and just accept this as groundwork for now, while not going as far as putting it in CI. If we do any more work on 0.23 we'll push for full pytest support.

File: tests/test_astronomer_commander.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        show_only=["charts/astronomer/templates/commander/commander-deployment.yaml"],
+    )
+
+    assert len(docs) == 1
+    doc = docs[0]
+    assert doc["kind"] == "Deployment"
+    assert doc["apiVersion"] == "apps/v1"
+    assert doc["metadata"]["name"] == "RELEASE-NAME-commander"
+    
+    envVarNames = jmespath.search("spec.template.spec.containers[0].env[*].name", doc)
```
Comment (danielhoherd) [inline tests/test_astronomer_commander.py:53]: Same reshaping question applies here.

File: tests/test_astronomer_commander.py (hunk 2 above)
Comment (danielhoherd) [reply tests/test_astronomer_commander.py:32]: Also I added a length assertion here because we are making assumptions about what `[0]` is.

**Review by danielhoherd — APPROVED** [review]

**Pull request merged**